packages/web-config-server/src/apiV1/dataBuilders/helpers/calculateOperationForAnalytics.js

```diff
@@ -41,20 +43,53 @@ const performSingleAnalyticOperation = (analytics, config) => {
   } else if (filteredAnalytics.length === 0) {
     return NO_DATA_AVAILABLE;
   }
-  return OPERATORS[operator](filteredAnalytics[0].value, config);
+  return OPERATORS[operator](filteredAnalytics[0].value, config, models);
 };
 
-const sumDataValues = (analytics, dataValues, filter = {}) => {
-  let sum; // Keep sum undefined so that if there's no data values then we can distinguish between No data and 0
+const filterAnalytics = async (analytics, filter, models) => {
   const { organisationUnit: organisationUnitFilter } = filter;
+  const { parent: parentFilter } = organisationUnitFilter;
+
+  if (!(organisationUnitFilter || parentFilter)) return analytics;
+
+  // if filtering on parent lets pre-build a orgUnit-parent map
+  const orgUnitParentMap = parentFilter
+    ? await buildOrgUnitParentMapForAnalytics(analytics, models)
+    : {};
+
+  return analytics.filter(a => {
+    if (parentFilter) {
+      const parent = orgUnitParentMap[a.organisationUnit];
+      const parentSatisfiesConditions = Object.entries(parentFilter).every(
+        ([field, thisFilter]) => {
+          return checkValueSatisfiesCondition(parent[field], thisFilter);
+        },
+      );
+      if (!parentSatisfiesConditions) {
+        return false;
+      }
+    }
 
-  analytics.forEach(({ dataElement, value, organisationUnit }) => {
     if (
       organisationUnitFilter &&
-      !checkValueSatisfiesCondition(organisationUnit, organisationUnitFilter)
+      !(
+        isPlainObject(organisationUnitFilter) &&
+        Object.keys(organisationUnitFilter).length === 1 &&
+        Object.keys(organisationUnitFilter)[0] === 'parent'
+      ) &&
+      !checkValueSatisfiesCondition(a.organisationUnit, organisationUnitFilter)
```

Comment:
Filtering out parent filter from organisationUnitFilter else missing 'operator' when it tries to check as parent filter is a child of organisationUnitFilter. Maybe a cleaner way to do this, could check on simple value (other reports currently only using simple code) but this allows eg exclude code

Reply:
Maybe filter out to check function would be better i.e. `!isParentFilter(organisationUnitFilter)`

Reply:
There is:

```js
 const { parent: parentFilter, orgUnitFilter } = filter.organisationUnit || {};
```

And then you can simply check for existence of `parentFilter` and `orgUnitFilter` before the corresponding `checkValueSatisfiesCondition` calls. 

We could probably make this even simpler by refactoring `checkValueSatisfiesCondition` to skip checking if `condition` is `undefined`, but this involves a bit of regression risk so we can skip it for now.

Reply:
Ah yes, so parent: parentFilter does the job of filtering out parent from the filter. I like it.

Reply:
(I just updated my comment above, I forgot the `...` on `orgUnitFilter`

Reply:
> condition is undefined worries me a bit if it is a legitimate error.

Good point... Probably the reason we haven't gone this way

Reply:
`const { parent: parentFilter, ...orgUnitFilter } = filter.organisationUnit || {};` splits organisationUnit into an array of char if string. May have to break it out into a couple of steps.

Reply:
Oh yes, you can change the approach according to the type of `filter.organisationUnit`

Reply:
Added check for empty condition object in `checkValueSatisfiesCondition` I think that is safe and a fair assumption.

Reply:
Very good call, will simplify things here!